src/compute-client/src/controller/instance.rs

```diff
@@ -1051,7 +1057,7 @@ where
             index_imports: dataflow.index_imports,
             objects_to_build: dataflow.objects_to_build,
             index_exports: dataflow.index_exports,
-            as_of: dataflow.as_of,
+            as_of: dataflow.as_of.clone(),
```

Comment:
What makes the `clone` necessary?

Reply:
The `as_ref` in the
```
        let as_of = dataflow
            .as_of
            .as_ref()
            .ok_or(DataflowCreationError::MissingAsOf)?;
```
at the beginning. Also, I didn't want to change the `as_ref` to a `clone` there, because I was worried that if we have 2 `as_of`s (one in the dataflow and one in the local variable `as_of`), then we might accidentally add some code that modifies one of these, but then we use the other one somewhere.